zerver/lib/create_user.py

```diff
@@ -154,6 +157,14 @@ def create_user(
     force_date_joined: Optional[datetime] = None,
     enable_marketing_emails: Optional[bool] = None,
 ) -> UserProfile:
+    if bot_type is None:
+        realm_user_default = RealmUserDefault.objects.get(realm=realm)
```

Comment:
Should we fetch this outside the `if` statement? I know it wastes a query when creating bots, but I don't like the control flow structure where we fetch it here conditionally and use it under the same condition below; just feels hard to analyze.

Reply:
Yeah, I did this for saving one query but I agree it is better to move the query before if statement. Updated it.